services/security/authentication/casigned/casigned.go

```diff
@@ -37,6 +44,47 @@ func (c *client) Login(ctx context.Context, group string, identity *security.Ide
 	return &response.Token, nil
 }
 
+func RenewRoutine(group, server string) {
+	renewalAttempt := 0
+	// Waiting for a few seconds to avoid spamming short-lived sdk user
+	time.Sleep(time.Second * 5)
+	for {
+		wssdConfig := auth.WssdConfig{}
+		err := marshal.FromJSONFile(auth.GetWssdConfigLocation(), &wssdConfig)
+		if err != nil {
+			fmt.Printf("Failed to open config file in location %s: %v\n", auth.GetWssdConfigLocation(), err)
+			panic("Failed to open config file")
+		}
+
+		sleepTime, err := renewTime(wssdConfig.ClientCertificate)
+		if err != nil {
+			fmt.Printf("Failed to find sleep time for cert %v \n", err)
+			panic("Failed to find renew time for certificate")
+		}
+		fmt.Printf("Waiting for %v to renew cert\n", sleepTime)
```

Comment:
[nitpick] the problem with Printf statements is that if a command line tool uses the SDK, its output will suddenly be "polluted" with all the logging coming from the SDK. Is there a way we could pass a logger from the Client to the routine? Feel free to push back and we can handle this later.

Reply:
I thought of introducing lambda for logging, but felt printing would be simpler solution as the renewal is not a very frequent operation. I can introduce loggers if needed in another iteration

Reply:
I think you need the LoginWithConfig to take in a ctx so you can log these events.  Our life will be miserable if we do not log these.
LoginWithConfig(ctx context.Context, ....)  and pass that to go RenewRouting(ctx context.Context, ...)
Then you will be able to log all your printfs